python/mxnet/gluon/estimator/event_handler.py

```diff
@@ -0,0 +1,100 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=wildcard-import
+"""Gluon EventHandlers for Estimators"""
+
+__all__ = ['EventHandler', 'LoggingHandler']
+import logging
+import os
+import time
+
+
+class EventHandler(object):
+    def __init__(self, estimator):
+        self._estimator = estimator
+
+    def train_begin(self):
+        pass
+
+    def train_end(self):
+        pass
+
+    def batch_begin(self):
+        pass
+
+    def batch_end(self):
+        pass
+
+    def epoch_begin(self):
+        pass
+
+    def epoch_end(self):
+        pass
+
+
+class LoggingHandler(EventHandler):
+    """Basic Logging Handler that applies to every Gluon estimator by default.
+    TODO: add doc
+    """
+
+    def __init__(self, estimator, log_name=None, file_name=None, file_location=None, ):
+        super(LoggingHandler, self).__init__(estimator)
+        log_name = log_name or 'Gluon Estimator'
+        self.logger = logging.getLogger(log_name)
+        self.logger.setLevel(logging.INFO)
+        streamhandler = logging.StreamHandler()
+        self.logger.addHandler(streamhandler)
+        # save logger to file only if file name or location is specified
+        if file_name or file_location:
+            file_name = file_name or log_name or 'estimator_log'
+            file_location = file_location or './'
+            filehandler = logging.FileHandler(os.path.join(file_location, file_name))
+            self.logger.addHandler(filehandler)
+
+    def train_begin(self):
+        pass
```

Comment:
might be a good idea to log all the hyper params and defaults used here

Reply:
will be added with the change on train_stats in a follow up PR